src/Control/Monad/Free.hs

```diff
@@ -53,14 +51,24 @@ import Control.Monad.State.Class
 import Control.Monad.Error.Class
 import Control.Monad.Cont.Class
 import Data.Functor.Bind
+import Data.Functor.Classes
 import Data.Foldable
 import Data.Profunctor
 import Data.Traversable
 import Data.Semigroup.Foldable
 import Data.Semigroup.Traversable
 import Data.Data
 import Prelude hiding (foldr)
-import Prelude.Extras
+
+import qualified Prelude.Extras as PE
+
+#ifdef LIFTED_FUNCTOR_CLASSES
+#if !MIN_VERSION_base(4,8,0)
+import Data.Monoid (mappend)
+#endif
+#else
+import Data.Functor.Classes.Lift
+#endif
```

Comment:
You might consider reexporting `Data.Functor.Classes` and `Data.Monoid` from `Data.Functor.Classes.Lift` to reduce the amount of boilerplate imports you'll need in each module.